src/components/sidebar/components.test-pw.tsx

```diff
@@ -0,0 +1,216 @@
+  const onCancel = () => {
+    setIsOpen(false);
+  };
+  return (
+    <>
+      <Button onClick={() => setIsOpen(true)}>Open sidebar</Button>
+      <Sidebar {...args} aria-label="sidebar" open={isOpen} onCancel={onCancel}>
+        <Box mb={2}>
+          <Button buttonType="primary">Test</Button>
+          <Button buttonType="secondary" ml={2}>
+            Last
+          </Button>
+        </Box>
+        <Box mb="3000px">Main content</Box>
+      </Sidebar>
+    </>
+  );
+};
+
+export const SidebarComponent = (props: Partial<SidebarProps>) => {
+  return (
+    <>
+      <Sidebar
+        aria-label="sidebar"
+        open
+        position="right"
+        size="medium"
+        {...props}
+      >
+        <Box mb={2}>
+          <Button buttonType="primary">Test</Button>
+          <Button buttonType="secondary" ml={2}>
+            Last
+          </Button>
+        </Box>
+        <Box mb="3000px">Main content</Box>
+      </Sidebar>
+    </>
+  );
+};
+
+export const SidebarComponentWithOnCancel = (props: Partial<SidebarProps>) => {
+  const [isOpen, setIsOpen] = useState(true);
+  const handleOnCancel = () => {
+    setIsOpen(false);
+  };
+  return (
+    <>
+      <Sidebar
+        aria-label="sidebar"
+        open={isOpen}
+        position="right"
+        size="medium"
+        onCancel={handleOnCancel}
+        {...props}
+      >
+        <Box mb={2}>
+          <Button buttonType="primary">Test</Button>
+          <Button buttonType="secondary" ml={2}>
+            Last
+          </Button>
+        </Box>
+        <Box mb="3000px">Main content</Box>
+      </Sidebar>
+    </>
+  );
+};
+
+export const SidebarComponentWithHeading = (props: Partial<SidebarProps>) => {
+  return (
+    <>
+      <Sidebar
+        aria-label="sidebar"
+        open
+        position="right"
+        size="medium"
+        header={<Typography variant="h3">Sidebar Header</Typography>}
+        {...props}
+      >
+        <Box mb={2}>
+          <Button buttonType="primary">Test</Button>
+          <Button buttonType="secondary" ml={2}>
+            Last
+          </Button>
+        </Box>
+        <Box mb="3000px">Main content</Box>
+      </Sidebar>
+    </>
+  );
+};
+
+export const SidebarBackgroundScrollTestComponent = () => {
+  return (
+    <Box height="2000px" position="relative">
+      <Box
+        data-element="test-box"
+        height="100px"
+        id="bottom-box"
+        position="absolute"
+        bottom="0px"
+      >
+        I should not be scrolled into view
+      </Box>
+      <Sidebar open onCancel={() => {}}>
+        <Textbox label="textbox" />
+      </Sidebar>
+    </Box>
+  );
+};
+
+export const SidebarBackgroundScrollWithOtherFocusableContainers = () => {
+  const toast1Ref = useRef(null);
+  const toast2Ref = useRef(null);
+  return (
+    <Box height="2000px" position="relative">
+      <Box
+        data-element="test-box"
+        height="100px"
+        id="bottom-box"
```

Comment:
**suggestion** (non-blocking): as above, the `id` can be removed from this example